src/language-js/print/function-parameters.js

```diff
@@ -223,6 +225,47 @@ function shouldGroupFunctionParameters(functionNode, returnTypeDoc) {
   );
 }
 
+/**
+ * The "decorated function" pattern.
+ * The arrow function should be kept hugged even if its signature breaks.
+ *
+ * ```
+ * const decoratedFn = decorator(param1, param2)((
+ *   ...
+ * ) => {
+ *   ...
+ * });
+ * ```
+ * @param {AstPath} path
+ */
+function isDecoratedFunction(path) {
+  return path.match(
+    (node) =>
+      node.type === "ArrowFunctionExpression" &&
+      node.body.type === "BlockStatement",
+    (node, name) => {
+      if (
+        node.type === "CallExpression" &&
+        name === "arguments" &&
+        node.arguments.length === 1 &&
+        node.callee.type === "CallExpression"
+      ) {
+        const decorator = node.callee.callee;
+        return (
+          decorator.type === "Identifier" ||
+          (decorator.type === "MemberExpression" &&
+            !decorator.computed &&
+            decorator.object.type === "Identifier" &&
+            decorator.property.type === "Identifier")
+        );
+      }
+      return false;
+    },
+    (node, name) => node.type === "VariableDeclarator" && name === "init",
+    (node) => node.kind === "const" && node.declarations.length === 1
```

Comment:
Similar to @sosukesuzuki I don't see why this feature should only work for `const` declarations (even if I personally would only use it with that). 

I also think it should work for `export default`, right?

Reply:
`export default` is a nice catch. As for `const`, I just want to minimize false positives. We can change this later if somebody requests that.